### PR TITLE
Provide Compact API

### PR DESCRIPTION
**Pull request by licanhua**
Add **UseCompactResources** to **XamlControlsResources**. eg:

This is how customer consume winui:
```
<Application>
    <Application.Resources>
        <XamlControlsResources xmlns="using:Microsoft.UI.Xaml.Controls" /> 
    </Application.Resources>
</Application>
```

This is how customer enable compact
```
<Application>
    <Application.Resources>
        <XamlControlsResources xmlns="using:Microsoft.UI.Xaml.Controls"  UseCompactResources="True"/> 
    </Application.Resources>
</Application>
```

### CODE REVIEW

File: active/DensityStyle/UseCompactResources.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+(http://docs.microsoft.com/windows/uwp/design/controls-and-patterns/password-box). -->
+UseCompactResources property is added XamlControlsResources to allow customer to enable Compact.
+
+# Examples
+
+```
+    runtimeclass XamlControlsResources : Windows.UI.Xaml.ResourceDictionary
+    {
+        ...
+        [WUXC_VERSION_PREVIEW]
```
Comment (jevansaks) [inline active/DensityStyle/UseCompactResources.md]:
```suggestion
```

File: active/DensityStyle/UseCompactResources.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
+# Background
+Compact feature is described in [Suggestion: Add Compact Dictionary to WinUI 2.1](https://github.com/Microsoft/microsoft-ui-xaml/issues/98).
```
Comment (MikeHillberg) [inline active/DensityStyle/UseCompactResources.md]:
```suggestion
Xaml ships with a set of resources (frequently called "generic.xaml") which define how controls appear.
The [WinUI](https://docs.microsoft.com/en-us/uwp/toolkits/winui/) package, which ships
several Xaml controls as a Nuget package, also ships an updated set of resources, in the form
of the 
[XamlControlsResources](https://docs.microsoft.com/en-us/uwp/api/microsoft.ui.xaml.controls.xamlcontrolsresources)
API. You bring this into your app by adding it to your Application.Resources:

````xaml
<Application>
    <Application.Resources>
        <XamlControlsResources xmlns="using:Microsoft.UI.Xaml.Controls"/> 
    </Application.Resources>
</Application>
````xaml

This spec adds a new property to XamlControlResources that updates the resources to make
the controls more "dense". For example, reducing the default spacing between controls.
Overall this allows for more UI to fit into the same screen space.
Density is described in
[Suggestion: Add Compact Dictionary to WinUI 2.1](https://github.com/Microsoft/microsoft-ui-xaml/issues/98).
```

File: active/DensityStyle/UseCompactResources.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
+# Background
+Compact feature is described in [Suggestion: Add Compact Dictionary to WinUI 2.1](https://github.com/Microsoft/microsoft-ui-xaml/issues/98).
+This provides API to enable Compact in application level.
```
Comment (MikeHillberg) [inline active/DensityStyle/UseCompactResources.md]:
```suggestion
```

File: active/DensityStyle/UseCompactResources.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
+# Background
+Compact feature is described in [Suggestion: Add Compact Dictionary to WinUI 2.1](https://github.com/Microsoft/microsoft-ui-xaml/issues/98).
+This provides API to enable Compact in application level.
+
+# Description
+<!-- Use this section to provide a brief description of the feature.
```
Comment (MikeHillberg) [inline active/DensityStyle/UseCompactResources.md]: Could you delete all these template comments?

File: active/DensityStyle/UseCompactResources.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+# Description
+<!-- Use this section to provide a brief description of the feature.
+For an example, see the introduction to the PasswordBox control 
+(http://docs.microsoft.com/windows/uwp/design/controls-and-patterns/password-box). -->
+UseCompactResources property is added XamlControlsResources to allow customer to enable Compact.
+
+# Examples
+
+```
+    runtimeclass XamlControlsResources : Windows.UI.Xaml.ResourceDictionary
```
Comment (MikeHillberg) [inline active/DensityStyle/UseCompactResources.md]: This isn't an example, this is the API Details. Could you move this down to that section?

File: active/DensityStyle/UseCompactResources.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    {
+        ...
+        {
+            Boolean UseCompactResources{ get; set; };
+            static Windows.UI.Xaml.DependencyProperty UseCompactResourcesProperty{ get; };
```
Comment (MikeHillberg) [inline active/DensityStyle/UseCompactResources.md]: Do we need this to be a DP?

File: active/DensityStyle/UseCompactResources.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+UseCompactResources property is added XamlControlsResources to allow customer to enable Compact.
+
+# Examples
+
+```
+    runtimeclass XamlControlsResources : Windows.UI.Xaml.ResourceDictionary
+    {
+        ...
+        {
+            Boolean UseCompactResources{ get; set; };
```
Comment (MikeHillberg) [inline active/DensityStyle/UseCompactResources.md]: Can I set this at any time? Or is it like RequestedTheme that has to be set really early?

File: active/DensityStyle/UseCompactResources.md [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+
+# API Notes
+
+
+# API Details
+<!-- The exact API, in MIDL3 format (https://docs.microsoft.com/en-us/uwp/midl-3/) -->
+## Class: XamlControlsResources 
+| Member Name | Description |
+|:- |:--|
+| UseCompactResources | True to enable Compact Resources. False when parameter is not set |
```
Comment (MikeHillberg) [inline active/DensityStyle/UseCompactResources.md]:
```suggestion
| UseCompactResources | True to enable Compact Resources. Defaults to false |
```

File: active/DensityStyle/UseCompactResources.md [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+        {
+            Boolean UseCompactResources{ get; set; };
+            static Windows.UI.Xaml.DependencyProperty UseCompactResourcesProperty{ get; };
+        }
+    }
+```
+
+# Remarks
+```
+<Application>
```
Comment (MikeHillberg) [inline active/DensityStyle/UseCompactResources.md]: This is an example, can you move it up the Examples section?

File: active/DensityStyle/UseCompactResources.md (hunk 7 above)
Comment (MikeHillberg) [inline active/DensityStyle/UseCompactResources.md]: This is phrased as an imperative, which we use for methods, not properties.

How about AreCompactResourcesEnabled? IsCompact?

Comment (licanhua) [reply active/DensityStyle/UseCompactResources.md]: No. I don't expect customer to change this dynamically.
it should be before UI is showed up. I see some weird crashes if I switch this flag multiple times when page is showed up.

File: active/DensityStyle/UseCompactResources.md (hunk 6 above)
Comment (licanhua) [reply active/DensityStyle/UseCompactResources.md]: @jevansaks  As I know, we always paired them in idl.

Comment (jevansaks) [reply active/DensityStyle/UseCompactResources.md]: Mike's asking if this property needs to be backed by a DP. I don't think so, but our general policy is that properties on controls are dependency properties unless there's a reason for them not to be.

This isn't a control but is a DO. What should the policy be?

File: active/DensityStyle/UseCompactResources.md (hunk 7 above)
Comment (licanhua) [reply active/DensityStyle/UseCompactResources.md]: @LucasHaines How do you think about AreCompactResourcesEnabled? I prefer AreCompactResourcesEnabled.

**Comment by licanhua** [comment]
We will not release it in 2.1 framework package, but in prerelease with preview tag.

**Pull request closed without merging**